ml-agents/mlagents/trainers/tests/test_barracuda_converter.py

```diff
@@ -24,3 +29,58 @@ def test_barracuda_converter():
 
     # cleanup
     os.remove(tmpfile)
+
+
+@pytest.fixture
+def dummy_config():
+    return yaml.safe_load(
+        """
+        trainer: ppo
+        batch_size: 32
+        beta: 5.0e-3
+        buffer_size: 512
+        epsilon: 0.2
+        hidden_units: 128
+        lambd: 0.95
+        learning_rate: 3.0e-4
+        max_steps: 5.0e4
+        normalize: true
+        num_epoch: 5
+        num_layers: 2
+        time_horizon: 64
+        sequence_length: 64
+        summary_freq: 1000
+        use_recurrent: false
+        normalize: true
+        memory_size: 8
+        curiosity_strength: 0.0
+        curiosity_enc_size: 1
+        summary_path: test
+        model_path: test
+        reward_signals:
+          extrinsic:
+            strength: 1.0
+            gamma: 0.99
+        """
+    )
+
+
+@pytest.mark.parametrize("discrete", [True, False], ids=["discrete", "continuous"])
+@pytest.mark.parametrize("visual", [True, False], ids=["visual", "vector"])
+@pytest.mark.parametrize("rnn", [True, False], ids=["rnn", "no_rnn"])
+def test_policy_conversion(dummy_config, tmpdir, rnn, visual, discrete):
+    tf.reset_default_graph()
+    dummy_config["summary_path"] = str(tmpdir)
+    dummy_config["model_path"] = os.path.join(tmpdir, "test")
+    policy = create_policy_mock(
+        dummy_config, use_rnn=rnn, use_discrete=discrete, use_visual=visual
+    )
+    policy.save_model(1000)
+    settings = SerializationSettings(
+        policy.model_path, os.path.join(tmpdir, policy.brain.brain_name)
+    )
+    export_policy_model(settings, policy.graph, policy.sess)
```

Comment:
Should we do this in `_check_environment_trains()` too?

Reply:
I think it's already done in `_check_environment_trains` right? The function seems to spit out an .nn to a temp directory

Reply:
FWIW I logged https://jira.unity3d.com/browse/MLA-740 to make sure we test the ONNX path for these (although not all CI configs can handle it since onnx+tf2 don't work yet).

Reply:
I guess the serialization is done in at the end of `start_learning()` 👍